main.cpp

```diff
@@ -91,15 +91,118 @@ good to go!
  Wait for my code review.
  */
 
+
 #include <iostream>
-int main() 
+#include <math.h> 
+
+struct FloatType
+{
+    float add(float, float);
+    float subtract(float, float);
+    float multiply(float, float);
+    float divide(float, float);
+};
+    
+float FloatType::add(float lhs, float rhs) 
+{
+     return lhs + rhs; 
+}
+
+float FloatType::subtract(float lhs, float rhs)
+{
+    return lhs - rhs;    
+}
+
+float FloatType::multiply(float lhs, float rhs)
+{
+    return lhs * rhs;
+}
+
+float FloatType::divide(float lhs, float rhs )
+{
+    if( rhs == 0.f )  
+    { 
+        printf("\nwarning, floating point division by zero returns 'inf' !\n");
+        return lhs/rhs;
+    }
+    
+    return lhs / rhs;
+}
```

Comment:
`return lhs / rhs` is written twice, and every logic path in this function leads to that line of code.
it does not need to be written twice, and does not need to be written inside the `if() { }` block

Don't use `printf`
use `std::cout` instead.

- [ ] revise